concourse/pipelines/templates/jobs/build-tpl.yml

```
@@ -3,7 +3,10 @@
  # Expects a dictionary with the name 'x' to be set as defined in macros.j2 file
  #}
 
-- name: build-pxf-gp[[x.gp_ver]]-[[x.build_platform]]
+{% set name = 'build-pxf-gp' ~ x.gp_ver ~ '-' ~ x.test_platform %}
+{% do x.update({'job_name': name}) %}
+
+- name: [[name]]
```

if we update the `x.job_name`, shall we use `[[x.job_name]]`?
```suggestion
- name: [[x.job_name]]
```

What's the benefits of directly using the `[[name]]` here?

Again the `name` is locally defined at line 6. Shall we distinguish between local variable vs. a passed-in parameter?

One more thing, if we use `[[name]]` here, I would put into the header commend to indicate this template actually depends on the input variable `name`, but in our case, we only generate the name locally based on the input `x`.